chat.go

```diff
@@ -69,18 +69,31 @@ type FunctionCall struct {
 	Arguments string `json:"arguments,omitempty"`
 }
 
+type ChatCompletionResponseFormatType string
+
+const (
+	ChatCompletionResponseFormatTypeJSONObject ChatCompletionResponseFormatType = "json_object"
+	ChatCompletionResponseFormatTypeText       ChatCompletionResponseFormatType = "text"
+)
+
+type ChatCompletionResponseFormat struct {
+	Type ChatCompletionResponseFormatType `json:"type"`
+}
+
 // ChatCompletionRequest represents a request structure for chat completion API.
 type ChatCompletionRequest struct {
-	Model            string                  `json:"model"`
-	Messages         []ChatCompletionMessage `json:"messages"`
-	MaxTokens        int                     `json:"max_tokens,omitempty"`
-	Temperature      float32                 `json:"temperature,omitempty"`
-	TopP             float32                 `json:"top_p,omitempty"`
-	N                int                     `json:"n,omitempty"`
-	Stream           bool                    `json:"stream,omitempty"`
-	Stop             []string                `json:"stop,omitempty"`
-	PresencePenalty  float32                 `json:"presence_penalty,omitempty"`
-	FrequencyPenalty float32                 `json:"frequency_penalty,omitempty"`
+	Model            string                       `json:"model"`
+	Messages         []ChatCompletionMessage      `json:"messages"`
+	MaxTokens        int                          `json:"max_tokens,omitempty"`
+	Temperature      float32                      `json:"temperature,omitempty"`
+	TopP             float32                      `json:"top_p,omitempty"`
+	N                int                          `json:"n,omitempty"`
+	Stream           bool                         `json:"stream,omitempty"`
+	Stop             []string                     `json:"stop,omitempty"`
+	PresencePenalty  float32                      `json:"presence_penalty,omitempty"`
+	ResponseFormat   ChatCompletionResponseFormat `json:"response_format,omitempty"`
```

Comment:
This is optional. So it should have been a pointer: `*ChatCompletionResponseFormat`. Now I get:

```
status code: 400, message: '' is not one of ['json_object', 'text'] - 'response_format.type'
```